filesystem/apps/zfs/nvpair.rs

```diff
@@ -0,0 +1,97 @@
+enum DataType {
+    Unknown = 0,
+    Boolean,
+    Byte,
+    Int16,
+    Uint16,
+    Int32,
+    Uint32,
+    Int64,
+    Uint64,
+    String, // TODO: What to name this string type?
+    ByteArray,
+    Int16Array,
+    Uint16Array,
+    Int32Array,
+    Uint32Array,
+    Int64Array,
+    Uint64Array,
+    StringArray,
+    Hrtime,
+    Nvlist, // TODO: What to name this ?
+    NvlistArray,
```

Comment:
Could you change:

`Hrtime` -> `HrTime`
`Nvlist` -> `NvList`
`NvListArray` -> `NvListArray`

Seems more consistent with the rest of the naming